File: 0983-validate-stack-sequences/0983-validate-stack-sequences.java
class Solution {
    public boolean validateStackSequences(int[] pushed, int[] popped) {
        int i =0 , j = 0;
        Stack<Integer> stack = new Stack<>();
        while(i < pushed.length){
          stack.push(pushed[i]);

          while(j < popped.length &&  !stack.isEmpty() && popped[j] == stack.peek()){
            stack.pop();
            j++;
          }
          i++;
        }
    return stack.isEmpty();
    }
}

File: 0983-validate-stack-sequences/0983-validate-stack-sequences.go
func validateStackSequences(pushed []int, popped []int) bool {
    i , j := 0 , 0
    stack := []int{}
    
    for i < len(pushed){
        stack = append(stack , pushed[i])
        i++
        for j < len(popped) && len(stack) > 0 && stack[len(stack) - 1] == popped[j]{
            stack = stack[:len(stack) -1]
            j++
        }
       
    }
      return len(stack) == 0
}


      